src/lib.rs

```diff
@@ -90,13 +85,15 @@ impl Drop for CmdLn {
 
 pub struct PsDecoder {
     raw: *mut bindings::ps_decoder_t,
+    #[allow(dead_code)]
+    config: CmdLn,
 }
 
 impl PsDecoder {
     pub fn init(config: CmdLn) -> Self {
-        let raw = unsafe { bindings::ps_init(config.into_raw()) };
+        let raw = unsafe { bindings::ps_init(config.raw) };
```

Comment:
You are right, `ps_init` increments reference counter: “The decoder retains ownership of the pointer config, so if you are not going to use it elsewere, you can free it.”